janus.c

```diff
@@ -1231,21 +1233,8 @@ int janus_process_incoming_request(janus_request *request) {
 				janus_plugin_result_destroy(result);
 				goto jsondone;
 			}
-			json_error_t error;
-			json_t *event = json_loads(result->content, 0, &error);
-			if(!event) {
-				JANUS_LOG(LOG_ERR, "[%"SCNu64"] Cannot send response from plugin (JSON error: on line %d: %s)\n", handle->handle_id, error.line, error.text);
-				ret = janus_process_error(request, session_id, transaction_text, JANUS_ERROR_PLUGIN_MESSAGE, "Plugin returned an invalid JSON response");
-				janus_plugin_result_destroy(result);
-				goto jsondone;
-			}
-			if(!json_is_object(event)) {
-				JANUS_LOG(LOG_ERR, "[%"SCNu64"] Cannot send response from plugin (JSON error: not an object)\n", handle->handle_id);
-				json_decref(event);
-				ret = janus_process_error(request, session_id, transaction_text, JANUS_ERROR_PLUGIN_MESSAGE, "Plugin returned an invalid JSON response");
-				janus_plugin_result_destroy(result);
-				goto jsondone;
-			}
+			/* Reference the content, as destroying the result instance will decref it */
+			json_incref(result->content);
```

Comment:
You mean, as opposed to an array for instance? Good point, I'll add it back, thanks!